test/local/ip_blocklist_tests.js

```diff
@@ -0,0 +1,207 @@
+        })
+    }
+  )
+
+  test(
+    name + ', returns true for empty ip',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains()
+          t.equal(result, true, 'return true for empty ip')
+          t.end()
+        })
+    }
+  )
+
+  test(
+    name + ', returns true for invalid ip',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains('notip')
+          t.equal(result, true, 'return true for invalid ip')
+          t.end()
+        })
+    }
+  )
+
+  test(
+    name + ', returns true for ip in blocklist',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains('1.93.0.224')
+          t.equal(result, true, 'return true for found ip')
+          t.end()
+        })
+    }
+  )
+
+  test(
+    name + ', returns true for ip in blocklist range',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains('0.0.0.1')
+          t.equal(result, true, 'return true for found ip')
+          t.end()
+        })
+    }
+  )
+
+  test(
+    name + ', returns false for ip not in blocklist',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains('12.34.32.1')
+          t.equal(result, false, 'return false for not found ip')
+          t.end()
+        })
+    }
+  )
+
+  test(
+    name + ', returns false for ip not in blocklist range',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains('3.0.0.0')
+          t.equal(result, false, 'return true for found ip')
+          t.end()
+        })
+    }
+  )
+})
+
+// Clear and Refresh test cases are not similar for IPBlocklist
+// and IPBlocklistManager
+test(
+  'IPBlocklist, clear blocklist',
+  function (t) {
+    var ipBlocklist = new IPBlocklist()
+
+    ipBlocklist.load(lists[0])
+      .then(function () {
+        ipBlocklist.clear()
+        t.deepEqual(ipBlocklist.ipsByPrefixLength, {}, 'empty ipsByPrefixLength object')
+        t.equal(ipBlocklist.prefixLengths.length, 0, 'empty ip prefix')
+        t.end()
+      })
+  }
+)
+
+test(
+  'IPBlocklist, refresh blocklist',
+  function (t) {
+    var ipBlocklist = new IPBlocklist()
+
+    ipBlocklist.load(lists[0])
+      .then(function () {
+        return ipBlocklist.refresh()
+      })
+      .then(function () {
+        t.end()
+      })
+      .catch(function (err) {
+        t.fail(err)
+        t.end()
+      })
+  }
+)
```

Comment:
It would be interesting to turn this into an explicit test that we correctly handle both additions and deletions from the list.  One option would be to:

* make a tempfile containing an initial version of a list
* load it and test that it works correctly
* overwrite the tempfile with new contents, removing some ips and adding others
* refresh it, and make sure it reflects the expected changes

Not sure how much work that would be to actually write in practice though :-)